src/ImageMetadata.jl

```diff
@@ -32,29 +33,38 @@ export
 Construct an image with `ImageMeta(A, props)` (for a properties dictionary
 `props`), or with `ImageMeta(A, prop1=val1, prop2=val2, ...)`.
 """
-mutable struct ImageMeta{T,N,A<:AbstractArray,P<:AbstractDict{String,Any}} <: AbstractArray{T,N}
+mutable struct ImageMeta{T,N,A<:AbstractArray,P<:AbstractDict{Symbol,Any}} <: AbstractArray{T,N}
     data::A
     properties::P
 
     function ImageMeta{T,N,A,P}(data::AbstractArray, properties::P) where {T,N,A,P}
```

Comment:
We should drop the `::P` from `properties::P`, thereby allowing one to use the inner constructor directly to coerce the properties dict to a specific type. Otherwise there isn't really any value in having this inner constructor.

Reply:
Are you suggesting
```julia
struct ImageMeta{T,N,A<:AbstractArray} <: AbstractArray{T,N}
    data::A
    properties

    function ImageMeta{T,N,A,P}(data::AbstractArray, properties::P) where {T,N,A,P<:AbstractDict{Symbol,Any}}
        new{T,N,A,P}(data, properties)
    end
end
```
or
```julia
struct ImageMeta{T,N,A<:AbstractArray,P} <: AbstractArray{T,N}
    data::A
    properties::P

    function ImageMeta{T,N,A,P}(data::AbstractArray, properties::P) where {T,N,A,P<:AbstractDict{Symbol,Any}}
        new{T,N,A,P}(data, properties)
    end
end
```
or
```julia
struct ImageMeta{T,N,A<:AbstractArray,P<:AbstractDict{Symbol,Any}} <: AbstractArray{T,N}
    data::A
    properties::P
end
```

I'd prefer the last one as it provides the most brevity without losing any functionality.

Reply:
I was talking about the signature of the arguments to the inner constructor, so none of these options are exactly what I meant. But I definitely like the last one best. (I'm guessing once we had a version that had to check something, but that's not happening now.)

If you ditch the inner constructor you can probably get rid of some of the outer constructors (rely on the defaults).